http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/RunnableRequest.java

```diff
@@ -111,13 +112,35 @@ private URI endpoint() {
     private void handleFailure(Supplier<String> msg, Throwable cause) {
         log.error(msg.get(), cause);
         runAndLogError("Exception thrown from AsyncResponseHandler",
-            () -> context.handler().exceptionOccurred(cause));
+                       () -> context.handler().exceptionOccurred(modifyHighBurstTrafficException(cause)));
         if (channel != null) {
             runAndLogError("Unable to release channel back to the pool.",
-                () -> closeAndRelease(channel));
+                           () -> closeAndRelease(channel));
         }
     }
 
+    private Throwable modifyHighBurstTrafficException(Throwable originalCause) {
+        String originalMessage = originalCause.getMessage();
+        String newMessage = null;
+
+        if (originalCause instanceof TimeoutException &&
+            originalMessage.contains("Acquire operation took longer then configured maximum time")) {
```

Comment:
Is there a better way of identifying these than the message? I'm worried they could change in future (or past) versions and it could be difficult to test.

If not, can we try to limit it to less of the message? Eg. if they fix the "then" typo, it would be nice if this didn't break.

Reply:
There is no specific exception types for these cases, so its difficult to identify other than the exception message. They have not changed these from 2 years (although doesn't guarantee they will not change). I can limit this message as this has a typo (then instead of than"